go.mod

```diff
@@ -1,114 +1,122 @@
 module github.com/mholt/caddy-dynamicdns
 
-go 1.18
```

Comment:
The choice to update all this was mainly because it was easier to get my tooling working this way. This is easy enough to exclude from the PR, wasn't sure if it would be welcome as a drive-by change.

(NixOS with flake.nix devShell, nixpkgs only has later versions of go which were unable to compile and run tests.)